homeassistant/components/media_player/__init__.py

```diff
@@ -878,16 +895,22 @@ async def _async_fetch_image(hass, url):
                     content_type = response.headers.get(CONTENT_TYPE)
                     if content_type:
                         content_type = content_type.split(";")[0]
-                    cache_images[url][CACHE_CONTENT] = content, content_type
 
         except asyncio.TimeoutError:
             pass
 
-        while len(cache_images) > cache_maxsize:
-            cache_images.popitem(last=False)
-
         return content, content_type
 
+    def get_browse_image_url(self, media_content_type, media_content_id, browse_image):
+        """Generate an url for a media browser image."""
+        base_url = get_url(self.hass)
+        url = (
+            f"{base_url}/api/media_player_proxy/{self.entity_id}"
+            f"?token={self.access_token}&media_content_type={media_content_type}"
```

Comment:
Use `YARL` to construct the query or else you might not encode the strings correctly.